modules/core/src/main/java/org/apache/ignite/internal/processors/cache/query/DistributedCacheQueryReducer.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache.query;
+
+import java.util.UUID;
+import org.apache.ignite.internal.IgniteInterruptedCheckedException;
+
+/**
+ * Reducer for distributed cache query.
+ */
+public interface DistributedCacheQueryReducer<T> extends CacheQueryReducer<T> {
+    /**
+     * Checks whether node with provided {@code nodeId} is a map node for the query.
+     * Note: if all pages were received from this node, then the method will return {@code false}.
+     *
+     * @param nodeId Node ID.
+     * @return {@code true} if node with provided {@code nodeId} is a map node for the query, {@code false} otherwise.
+     */
+    public boolean mapNode(UUID nodeId);
```

Comment:
```suggestion
    public boolean isMapNode(UUID nodeId);
```

Reply:
According to code style [1] we should not use "is" as prefix for internal API.

[1] https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines#CodingGuidelines-Gettersandsetters

Reply:
This is bad codestyle =)
The one have to read javadoc to understand what the method doing. 
It is unclear whether the method checks "the node is map-node" or method "make the node a map-node".
Maybe name "mappedToNode" be better? WDYT?

Reply:
WDYT if we violate this rule and name it `isMapNode`?

Reply:
Renamed to `remoteQueryNode`